Nukeeper.BitBucketLocal/BitbucketLocalRestClient.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using NuKeeper.Abstractions.Logging;
+using NuKeeper.BitBucketLocal.Models;
+
+namespace NuKeeper.BitBucketLocal
+{
+    public class BitbucketLocalRestClient
+    {
+        private readonly HttpClient _client;
+        private readonly INuKeeperLogger _logger;
+
+        public BitbucketLocalRestClient(HttpClient client, INuKeeperLogger logger, string username, string appPassword)
+        {
+            _client = client;
+            _logger = logger;
+
+            var byteArray = Encoding.ASCII.GetBytes($"{username}:{appPassword}");
+            _client.DefaultRequestHeaders.Authorization =
+                new AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteArray));
+            _client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+        }
+
+
+
+        private async Task<T> GetResourceOrEmpty<T>(string url)
+        {
+            var response = await _client.GetAsync(url).ConfigureAwait(false);
+
+            if (!response.IsSuccessStatusCode) return default;
```

Comment:
May want to look at the AzureDevops version of this, it handles errors in a more transparent way